### PR TITLE
Fix customer voucher "active" column in customer view page

### LINKED ISSUE #29361: BO - The status of the voucher is always disabled in the customer details page
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

This issue is previously reported here : #[28896](https://github.com/PrestaShop/PrestaShop/issues/28896)

The status of the voucher is always disabled in the customer details page

image

image

https://user-images.githubusercontent.com/92912932/184919170-cc58a5c5-abe9-439e-9644-1295e5e2bbb2.mp4

### Expected behavior

_No response_

### Steps to reproduce

1. BO > Catalog > Discounts > Create a new cart rule with enabled status
2. Go to catalog > Customers > View the attached customer
3. See error > The status is disabled in the Voucher section

### PrestaShop version(s) where the bug happened

8.0.x

### PHP version(s) where the bug happened

7.3

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | refer to https://github.com/PrestaShop/PrestaShop/issues/29361
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/29361
| Related PRs       | 
| How to test?      | refer to https://github.com/PrestaShop/PrestaShop/issues/29361
| Possible impacts? |

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Columns/Content/status.html.twig [hunk 1]
```diff
@@ -22,10 +22,8 @@
  * @copyright Since 2007 PrestaShop SA and Contributors
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  *#}
-{% set isValid = record[column.id] is same as(1) %}
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Columns/Content/status.html.twig:25]: I saw that change made recently to compare from string to int, idk why it needed to be done, but for the sake of simplicity now i made "truthy" unstrict comparison which should work for both cases ("active" column value comes as string "1", thats why this statement always failed)

**Review by yanmakouf — APPROVED** [review]
Ok to me. Maybe some test scenario would help to prevent breaking it in the future ? Since we don't know why this changed was introduced in the first place :)

**Comment by zuk3975** [comment]
> Ok to me. Maybe some test scenario would help to prevent breaking it in the future ? Since we don't know why this changed was introduced in the first place :)

I don't know what test could be written for this case :thinking:

**Review by kpodemski — APPROVED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Columns/Content/status.html.twig (hunk 1 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Columns/Content/status.html.twig:25]: That makes sense, apparently `same as` is the twig equivalent of `===`
Only thing that worries e is if `'0'` will correctly be casted to int and be considered falsy?
Maybe we can make this more explicit with: `{% set isActive = record[column.id]|format_number %}`

Seems like drupal has filters like `value|boolean` or `value|integer` but I guess they added their custom filters, it doesn't seem to be part of the native twig filters, too bad.

**Review by jolelievre — APPROVED** [review]
Thanks @zuk3975 

I still had a suggestion to make the test more explicit but I guess it's not blocking

**Review by florine2623 — APPROVED** [review]
Hello @zuk3975 ,

Thanks for the PR.

Tested on 8.0.x and launched the automated tests.

It is looking good. I created a few discounts and they are all displayed with the correct status.
In customer detail page the display of the status is correct:
Screenshot 2022-08-29 at 15 39 46
Checked if info matches in the Order > Discount page: it is OK.
Screenshot 2022-08-29 at 15 41 37

Also checked the customer account, only the enabled discount are displayed.
Screenshot 2022-08-29 at 15 42 32

It is QA ✅ !

**Comment by atomiix** [comment]
Thanks @zuk3975 & @florine2623

**Pull request merged**